### PR TITLE
wipe: fix build

### LINKED ISSUE #368675: Build failure: wipe
## Steps To Reproduce

1. build `wipe` from unstable (`github:nixos/nixpkgs/634fd46801442d760e09493a794c4f15db2d0cbb`).

## Build log

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/0fyqydvrgzq51lb8dbyfarx498pci3q4-wipe-2.3.1.tar.bz2
source root is wipe-2.3.1
setting SOURCE_DATE_EPOCH to timestamp 1257109891 of file wipe-2.3.1/wipe.h
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/m4bw4dag08g4gfsxrxiyk45h9128bfmi-fix-install.patch
patching file Makefile.in
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
configure flags: --prefix=/nix/store/kksykx8ipwcj2kjhyqxdh5pklf4g83rl-wipe-2.3.1
creating cache ./config.cache
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking build system type... x86_64-pc-linux-gnu
checking for gcc... gcc
checking whether the C compiler (gcc  ) works... no
configure: error: installation or configuration problem: C compiler cannot create executables.

```

</details>

## Additional context

I use `nixpkgs.hostPlatform.gcc.{arch,tune} = "znver4";`, which may be relevant.
Probably more so related to recent change for default gcc to 14.

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.5-xanmod1, NixOS, 25.05 (Warbler), 25.05.20241219.d70bd19`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/alfarel/.config/nix/nix.conf:/nix/store/v5pkh7wda19iw13vsc4ma7cr8sbhzxs1-stylix-kde-config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/alfarel/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/alfarel/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/alfarel/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/3vdv3p0nhfzz4k3xv5dfmn6qw653xc2z-lix-2.91.1/share`
 - nixpkgs: `/nix/store/8fwsiv0hd7nw1brkvka0jf1frk3m7qkr-source`

## Notify maintainers

@abbradar , but they haven't committed in 2 years, so not sure if they are still at all active

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by alfarelcynthesis**
Fix build for `wipe` after latest staging merge into unstable, courtesy of @paparodeo .
Closes: #368675 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by paparodeo — APPROVED** [review]

**Comment by paparodeo** [comment]
tho if you could remove `@paparodeo` from the commit that would be great -- otherwise I'll get lots of pings.

**Comment by alfarelcynthesis** [comment]
Sorry about that, didn't realise it'd do that.

**Review by paparodeo — APPROVED** [review]

**Comment by alfarelcynthesis** [comment]
I can't actually check that the `x86_64-darwin` build works now, but it should fix the error in [ofborg](https://logs.ofborg.org/?attempt_id=a17bfdb1-289f-45be-ab04-6e4adf26e093&key=nixos%2Fnixpkgs.368686).

File: pkgs/by-name/wi/wipe/package.nix [hunk 1]
```diff
@@ -13,6 +14,11 @@ stdenv.mkDerivation rec {
     sha256 = "180snqvh6k6il6prb19fncflf2jcvkihlb4w84sbndcv1wvicfa6";
   };
 
+  nativeBuildInputs = [ autoreconfHook ];
+
+  # ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
+  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isClang "-Wno-error=implicit-function-declaration";
```
Comment (paparodeo) [inline pkgs/by-name/wi/wipe/package.nix:20]:
```suggestion
  # fdatasync is undocumented on darwin with no header file which
  # breaks the build. use fsync instead
  configureFlags =
    lib.optional stdenv.hostPlatform.isDarwin "ac_cv_func_fdatasync=no";
```

Comment (alfarelcynthesis) [reply pkgs/by-name/wi/wipe/package.nix]: Thanks for all the help :)

**Comment by paparodeo** [comment]
@ofborg build wipe

**Review by paparodeo — APPROVED** [review]
built on x64 linux and aarch64 darwin

**Comment by alfarelcynthesis** [comment]
[Built on x86_64-darwin](https://logs.ofborg.org/?attempt_id=89fc0a6d-781b-4c22-8cb9-560e8e652f80&key=nixos%2Fnixpkgs.368686) now too.

**Pull request merged**